keymanapp/keyman
# chore(web): updates build product references, extracts CI test-build scripts 📜

This 📜 PR set is for some long-needed maintenance:  the Web project's files are being reorganized into a format more consistent with the other Keyman projects.  There's no new functionality; the end-goal is for everything to be stable and working as before (at the end of the 📜 PR chain).

Continues from #7556.

At this point in time, note that only "Test - 14.0" test config steps have been extracted.  There's more work to be done to also handle nightly / beta / stable test configs properly.

Now that all the `i`'s have been dotted and `t`'s crossed for properly reconnecting everything to our CI configurations (including on the CI server side!)...

### Other Notes

Re: the last would-be test defined below...

https://github.com/keymanapp/keyman/blob/c91875bbb19000cd2b54ad8637854cb680a04db8/web/src/tools/testing/recorder/recorder_KeyboardScripts.js#L1-L11

We'll need an extra configuration here to handle this page when hosted from the CI server; build artifacts aren't hosted from the site's root, but from a significantly-deep subfolder instead.  The stub-hosting part was "simple enough", but... the second half - loading from those stubs - may take some extra work.

![image](https://user-images.githubusercontent.com/25213402/204472732-5bd5f180-7e6a-425d-b788-ddf7bdf21104.png)


## User Testing

As this 📜 PR chain has shuffled a lot of Web files around, the main goal here is to ensure that everything is linked properly in its new location.

**TEST_ANDROID_LAUNCH**:  Verify that the Keyman for Android app launches successfully and displays the keyboard.
- Any Android device, emulated or real, will suffice.

**TEST_IOS_LAUNCH**:  Verify that the Keyman for iPhone and iPad app launches successfully and displays the keyboard.
- Any iOS device, simulated or real, will suffice.

**TEST_WEB_PREDICTIONS**:  Using Chrome **remote device emulation**...
- Navigate through the following links:
    - KeymanWeb Test Home (as linked [here](https://github.com/keymanapp/keyman/pull/7831#issuecomment-1328653844))
    - "View Keymanweb website-oriented manual test pages"
    - "Prediction UI testing"
- Click the "Type here" field.
- Verify that the OSK appears and that the predictive-text banner offers suggestions.

**TEST_ANDROID_HARNESS**:  Using Chrome **remote device emulation**...
- Navigate through the following links:
    - KeymanWeb Test Home (as linked [here](https://github.com/keymanapp/keyman/pull/7831#issuecomment-1328653844))
    - "View manual test pages for embedded-mode Keymanweb"
    - "Android test harness"
- Verify that an OSK appears at the bottom of the black box.
- Using the OSK, type `a` and verify that an `a` is emitted in the text area above the black box.

**TEST_WEB_SAMPLE_TOOLBAR**:  Using a standard desktop browser (and not emulating a touch device)...
- Navigate through the following links:
    - KeymanWeb Test Home (as linked [here](https://github.com/keymanapp/keyman/pull/7831#issuecomment-1328653844))
    - "View Keymanweb use samples"
    - "Full multilingual (compiled) Keymanweb"
- Verify that a "toolbar" of sorts appears on the page.
   - Wait a couple of seconds before proceeding; when additional options appear on the toolbar, proceed.
- Click the toolbar's "Languages" button and verify that a world map appears and that a list of languages is available for selection.

_**Known issue with the following test: needs extra configuration work.  May be delayed as a separate issue.**_

Skip this; I've ensured that the `test-bot` will too.

_SKIP_TEST_WEB_RECORDER_:  Using a standard desktop browser...
- Navigate through the following links:
    - KeymanWeb Test Home (as linked [here](https://github.com/keymanapp/keyman/pull/7831#issuecomment-1328653844))
    - "Use Keymanweb development tools."
    - "Test-sequence recorder"
- Verify that no alerts occur when loading the page.
- Scroll down to the "Input Recording" section, then click within the textbox immediately below "Record input here".
- A small menu should appear to the right-hand side of the textbox - use it to select "Simple Deadkeys (English)".
- Type `a` with the OSK, then `s` with your computer's hardware keyboard.
    - New data should show up in the big area to the right-hand side - new data for every keystroke.
    - You should see a `"type": "osk"` entry before any `"type": "key"` entry.
